Sources/GRPCHTTP2TransportNIOPosix/NIOSSL+GRPC.swift

```
@@ -0,0 +1,120 @@
+          )
+
+        case .pem:
+          let certificates = try NIOSSLCertificate.fromPEMFile(path).map {
+            NIOSSLCertificateSource.certificate($0)
+          }
+          certificateSources.append(contentsOf: certificates)
+        }
+      }
+    }
+    return certificateSources
+  }
+}
+
+extension NIOSSLPrivateKey {
+  fileprivate convenience init(
+    privateKeySource source: HTTP2ServerTransport.Config.TLS.PrivateKeySource
+  ) throws {
+    switch source.wrapped {
+    case .file(let path, let serializationFormat):
+      try self.init(
+        file: path,
+        format: NIOSSLSerializationFormats(serializationFormat)
+      )
+    case .bytes(let bytes, let serializationFormat):
+      try self.init(
+        bytes: bytes,
+        format: NIOSSLSerializationFormats(serializationFormat)
+      )
+    }
+  }
+}
+
+extension NIOSSLTrustRoots {
+  fileprivate init(_ trustRoots: HTTP2ServerTransport.Config.TLS.TrustRoots) throws {
+    switch trustRoots.wrapped {
+    case .pemFile(let path):
+      self = .file(path)
+    case .pemBytes(let bytes):
+      self = .certificates(try NIOSSLCertificate.fromPEMBytes(bytes))
+    }
+  }
+}
+
+extension TLSConfiguration {
+  internal init(_ tlsConfig: HTTP2ServerTransport.Config.TLS) throws {
+    let certificateChain = try NIOSSLCertificateSource.certificateChain(
+      from: tlsConfig.certificateChainSources
+    )
+    let privateKey = try NIOSSLPrivateKey(privateKeySource: tlsConfig.privateKeySource)
+
+    var tlsConfiguration = TLSConfiguration.makeServerConfiguration(
+      certificateChain: certificateChain,
+      privateKey: .privateKey(privateKey)
+    )
+    tlsConfiguration.minimumTLSVersion = .tlsv12
+    tlsConfiguration.certificateVerification =
+      tlsConfig.verifyClientCertificate ? .fullVerification : .none
+    tlsConfiguration.trustRoots = tlsConfig.trustRoots.map { try? NIOSSLTrustRoots($0) } ?? .default
```

Why do we suppress the error here?